### PR TITLE
[new release] mlpost-lablgtk and mlpost (0.9)

### PR DESCRIPTION
**Pull request by backtracking**
Library for adding mlpost graphics in lablgtk

- Project page: <a href="https://github.com/backtracking/mlpost">https://github.com/backtracking/mlpost</a>
- Documentation: <a href="https://backtracking.github.io/mlpost">https://backtracking.github.io/mlpost</a>

##### CHANGES:

- fixes floating-point error in extremum computation
  - special points are kept in empty from box
  - migrate to cairo2 and dune
  - added experimental export to pgf
  - added Arrows.mk_classic
  - added Helpers.box_point_line, point_box_line, box_loop, box_label_loop

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
ocaml-migrate-parsetree 2.0 is not supported
```
#=== ERROR while compiling mlpost.0.9 =========================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.11.1 | pinned(https://github.com/backtracking/mlpost/releases/download/0.9/mlpost-0.9.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/mlpost.0.9
# command              ~/.opam/4.11/bin/dune build -p mlpost -j 47 @install
# exit-code            1
# env-file             ~/.opam/log/mlpost-16-5f757a.env
# output-file          ~/.opam/log/mlpost-16-5f757a.out
### output ###
#          gcc src/ml_mlpost_ft.o
# In file included from ml_mlpost_ft.c:8:
# ml_mlpost_ft.c: In function 'ml_FT_Get_Name_Index':
# /home/opam/.opam/4.11/lib/ocaml/caml/mlvalues.h:265:24: warning: passing argument 2 of 'FT_Get_Name_Index' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
#  #define String_val(x) ((const char *) Bp_val(x))
#                        ~^~~~~~~~~~~~~~~~~~~~~~~~~
# ml_mlpost_ft.c:18:34: note: in expansion of macro 'String_val'
#                                   String_val (char_name));
#                                   ^~~~~~~~~~
# In file included from /usr/include/cairo/cairo-ft.h:47,
#                  from /home/opam/.opam/4.11/lib/cairo2/cairo_ocaml.h:151,
#                  from ml_mlpost_ft.c:9:
# /usr/include/freetype2/freetype/freetype.h:3903:34: note: expected 'FT_String *' {aka 'char *'} but argument is of type 'const char *'
#                       FT_String*  glyph_name );
#                       ~~~~~~~~~~~~^~~~~~~~~~
#       ocamlc customdoc/.img.objs/byte/img.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlc.opt -w -40 -g -bin-annot -I customdoc/.img.objs/byte -I /home/opam/.opam/4.11/lib/ocaml-migrate-parsetree -I /home/opam/.opam/4.11/lib/ocaml/compiler-libs -no-alias-deps -o customdoc/.img.objs/byte/img.cmo -c -impl customdoc/img.ml)
# File "customdoc/img.ml", line 2, characters 5-18:
# 2 | open OCaml_403.Ast
#          ^^^^^^^^^^^^^
# Error: Unbound module OCaml_403
```

File: packages/mlpost-lablgtk/mlpost-lablgtk.0.9/opam [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+depends: [
+  "ocaml" { >= "4.07.0" }
+  "bitstring" { >= "3.0.0" }
+  "dune" { >= "2.7.0" }
+  "mlpost"
```
Comment (kit-ty-kate) [inline packages/mlpost-lablgtk/mlpost-lablgtk.0.9/opam:35]: Are you planning on releasing `mlpost-lablgtk` alongside `mlpost`? If yes, I would highly suggest adding `{= version}` here.

**Comment by kit-ty-kate** [comment]
There is another error in:
```
#=== ERROR while compiling mlpost.0.9 =========================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.11.1 | pinned(https://github.com/backtracking/mlpost/releases/download/0.9/mlpost-0.9.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/mlpost.0.9
# command              ~/.opam/4.11/bin/dune build -p mlpost -j 47 @install
# exit-code            1
# env-file             ~/.opam/log/mlpost-15-b31ff6.env
# output-file          ~/.opam/log/mlpost-15-b31ff6.out
### output ###
#          gcc src/ml_mlpost_ft.o
# In file included from ml_mlpost_ft.c:8:
# ml_mlpost_ft.c: In function 'ml_FT_Get_Name_Index':
# /home/opam/.opam/4.11/lib/ocaml/caml/mlvalues.h:265:24: warning: passing argument 2 of 'FT_Get_Name_Index' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
#  #define String_val(x) ((const char *) Bp_val(x))
#                        ~^~~~~~~~~~~~~~~~~~~~~~~~~
# ml_mlpost_ft.c:18:34: note: in expansion of macro 'String_val'
#                                   String_val (char_name));
#                                   ^~~~~~~~~~
# In file included from /usr/include/cairo/cairo-ft.h:47,
#                  from /home/opam/.opam/4.11/lib/cairo2/cairo_ocaml.h:151,
#                  from ml_mlpost_ft.c:9:
# /usr/include/freetype2/freetype/freetype.h:3903:34: note: expected 'FT_String *' {aka 'char *'} but argument is of type 'const char *'
#                       FT_String*  glyph_name );
#                       ~~~~~~~~~~~~^~~~~~~~~~
#      img_doc customdoc/center.png,customdoc/circle.png,customdoc/ctr.png,customdoc/east.png,customdoc/ellipse.png,customdoc/halign.png,customdoc/hbox.png,customdoc/height.png,customdoc/hplace.png,customdoc/north.png,customdoc/north_east.png,customdoc/north_west.png,customdoc/patatoid.png,customdoc/rect.png,customdoc/round_rect.png,customdoc/shift.png,customdoc/south.png,customdoc/south_east.png,customdoc/south_west.png,customdoc/tex.png,customdoc/west.png,customdoc/width.png (exit 2)
# (cd _build/default/customdoc && ./img_doc.exe -cairo -png)
# Fatal error: exception Invalid_argument("One tex snippet contains an error")
```
Unless you want the documentation to be always installed, apart from fixing the compilation issue, I would also suggest to disable the documentation by default. Opam has a flag `{with-doc}` is case the user wants to have the documentation.

File: packages/mlpost-lablgtk/mlpost-lablgtk.0.9/opam (hunk 1 above)
Comment (backtracking) [reply packages/mlpost-lablgtk/mlpost-lablgtk.0.9/opam:35]: Thanks. Makes sense. I'll add this constraint.

**Comment by backtracking** [comment]
I don't understand why the doc is built but it should not. I'm investigating and I'll fix that.

**Comment by backtracking** [comment]
And thanks for fixing the `ocaml-migrate-parsetree` dependency.

**Pull request closed without merging**